### DISCUSSION TITLE
How to debug STM32F103C8T6 on Arduino IDE2?

### DISCUSSION THREAD

**Discussion by JingtaiChen** [stm32duino/Arduino_Core_STM32#2179] [Q&A]
How to debug STM32F103C8T6 on Arduino IDE2?

**Comment by fpistm** [comment]
You need  a STLink. Then build with symbol (-g) and optimize -Og. Flash and launch debug with the icon button.

**Comment by JingtaiChen** [comment]
When I click on this "添加配置Add configuration", 
<img width="213" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/111559992/97db13d3-10ad-4c76-bec5-f7d98ab4086d">
the next thing it pops up is the following json，How do I choose
<img width="487" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/111559992/2c0c545f-2357-4152-b937-de31967f1832">

**Reply by fpistm** [reply]
No. Don't use this button. It does not work. It is an Arduino IDE bug. Use the one on the top bar.

**Reply by JingtaiChen** [reply]
Do you mean this button
<img width="54" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/111559992/87ab6b1b-e54a-4ac2-b7ba-3e1345a5a127">
And then he checked the box, and I said yes
<img width="490" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/111559992/8e6f4668-dc48-404d-9064-117e640bb26c">
Then he reminded me of the mistake
<img width="491" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/111559992/4d1ba1bc-eb16-44ac-90ba-8d4ce24367e6">
Is this the reason why I did not select the STlinki port? However, stlink's port does not pop up in my port selection
<img width="249" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/111559992/e2830ebb-6c36-4d3a-84e0-411945341c6c">

**Comment by JingtaiChen** [comment]
I have used stlink to connect to my MAC and can download the program normally

**Comment by JingtaiChen** [comment]
<img width="845" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/111559992/818ee543-73ff-4987-a970-29b4d613fafa">

**Reply by fpistm** [reply]
Reset signal of the stlink is required.
